### ISSUE TITLE
Rework SCA Policy for Microsoft Windows 11 Enterprise

### ISSUE THREAD

**Issue by ooniagbi** [wazuh/wazuh#22195]
| Component | Action type           | Main Issue 
| --- | --- | ---
| SCA       | Rework |  #21930

## Main tasks

- [x] Use the latest CIS benchmark PDF from https://downloads.cisecurity.org/#/ 
- [x] Verify IDs numbers.
- [x] Verify texts are correct: Title, Description, Rationale and Remediation.
- [x] Verify Compliance: CIS, CIS_CSC.
- [x] Verify condition and rules:
  - [x] To Pass.
  - [x] To Fail.
- [x] Solve Related issues:
- https://github.com/wazuh/wazuh/issues/16479

### Syntax and semantic

- [x] a) ID of each policy must be contiguous.
- [x] b) The order and format set in [Documentation](https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/creating-custom-policies.html) must be respected.
- [x] c) YML must be valid to avoid errors.

### Content

- [x] a) Compare each check with its analog from CIS Benchmark.
- [x] b) Try maintaining each rule as similar as possible with the `Audit` section from the CIS check.
- [x] c) Check that the commands provide the expected output.
- [x] d) When a failure is discovered, check similar policies to avoid repetition of the issue.

### Unit testing

- [x] a) Output from `agent.log` after the SCA scan and a raw output of the result of the checks.
<details><summary>Tests results</summary>

```
2023/11/09 17:16:09 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win10_enterprise.yml'
2023/11/09 17:16:09 sca: INFO: Security Configuration Assessment scan finished. Duration: 81 seconds.
```

<p>

#### Analysisd (server or local)
analysisd.debug=2

#### Auth daemon debug (server)
authd.debug=0

#### Exec daemon debug (server, local, or Unix agent)
execd.debug=0

#### Monitor daemon debug (server, local, or Unix agent)
monitord.debug=0

#### Log collector (server, local or Unix agent)
logcollector.debug=0

#### Integrator daemon debug (server, local or Unix agent)
integrator.debug=0

#### Unix agentd
agent.debug=2

</p>
</details>

### Deployment

- [x] a) If the policy it's new, it must be added to the `sca.files` templates.
- [x] b) If the OS has many supported SCA policies, a policy must be set as the default policy. ([as example](https://github.com/wazuh/wazuh/blob/master/etc/templates/config/centos/sca.files))

### Documentation

- [x] a) Ensure documentation SCA list includes the created or updated SCA.

**Comment by jk-olaoluwa** [comment]
**Update 17/04/2024**
Section 18.9.5.1 - 18.9.5.7

**Comment by ooniagbi** [comment]
Currently working on other issues.

**Comment by ooniagbi** [comment]
This is still on hold because of higher-priority issues.

**Comment by dlehman83** [comment]
A couple suggestions for this rework and other future benchmark reworks.  
Include the profile in the title field
Split the profiles into separate files.  
This benchmark has 3 profiles with different combinations. 
L1 General purpose 
L2 High security 
BL Bluetooth.  

Allow someone to scan based on their chosen profile would be very helpful.  
The easiest way to achieve this may be to split L1 and Bluetooth into their own files, then L2 is of course L1 + L2.  
The other idea would be to include an additional profile property on each check, so it can be filtered / searched on.

**Comment by ooniagbi** [comment]
This is still on hold because of higher-priority issues.

**Comment by jk-olaoluwa** [comment]
## Update 05/08/2024

 - Preparing test environment

**Comment by jk-olaoluwa** [comment]
## Update 16/08/2024
 - Completed environment setup
 - Section 18.9.5.1 - 18.9.5.7

**Comment by jk-olaoluwa** [comment]
## Update 19/08/2024

 - Rework Section 18.9.5.1 - 18.9.20.1.14  - https://github.com/wazuh/wazuh/pull/22734/commits/0962acd2a621422b9d1d03c0d69c9643c227f52f

**Comment by jk-olaoluwa** [comment]
## Update 20/08/2024

 - Rework Section 18.9.20.1.14 - 18.9.26.2  - https://github.com/wazuh/wazuh/pull/22734/commits/da37abd20e780db1cb627294892bc7734401879e

**Comment by jk-olaoluwa** [comment]
## Update 20/08/2024

 - Rework Sections 18.9.26.2 - 18.9.51.1.2 - https://github.com/wazuh/wazuh/pull/22734/commits/a39cc643760cfbb78ac4d02697d129589bb7f9dd

**Comment by jk-olaoluwa** [comment]
## Update 22/08/2024

 - Rework Sections 18.9.26.2 - 18.9.51.1.2   - https://github.com/wazuh/wazuh/pull/22734/commits/266a4eb4204cb6f1ed79fbab5fd458463849e20b

**Comment by jk-olaoluwa** [comment]
## Update 23/08/2024

 - Rework Sections 18.9.51.1.2 - 18.10.9.1.13 - https://github.com/wazuh/wazuh/pull/22734/commits/2cbb3410107963e7ab349d46ab4f4a22d1bf7077

**Comment by jk-olaoluwa** [comment]
## Update 26/08/2024

 - Rework Sections 18.10.9.1.13 - 18.10.15.8 - https://github.com/wazuh/wazuh/pull/22734/commits/6b80c9bbd5d21c494d3731b12f0824a603f2f607

**Comment by jk-olaoluwa** [comment]
## Update 27/08/2024

 - Rework Sections 18.10.15.8 - 18.10.58.7 - https://github.com/wazuh/wazuh/pull/22734/commits/8f5130eecd7c3375c78b28ac405ac5bd08521a3b

**Comment by jk-olaoluwa** [comment]
## Update 28/08/2024

 - Rework Sections 18.10.58.7 - 18.10.92.2.2 - https://github.com/wazuh/wazuh/pull/22734/commits/6e29d247a3ee104b6ae09a0850a953578364ca61

**Comment by jk-olaoluwa** [comment]
## Update 28/08/2024

 - Rework Sections 18.10.92.2.2 - 19.7.44.2.1 - https://github.com/wazuh/wazuh/pull/22734/commits/35dff3f4fcab74a19b1b046290f35c49389571bc

**Comment by IsExec** [comment]
# Review Update - 03/09/2024

- Based on change log (17/62 completed) 🟢

**Comment by IsExec** [comment]
# Review Update - 04/09/2024

## Based on changelog (50/62 completed)


  - <details><summary>18.9.25.6 (L1) Ensure 'Password Settings: Password Age (Days)' is set to 'Enabled: 30 or fewer'. (Automated)
     🔴  </summary>

            - Id:             
            - Title:          🟢
            - Description:    🟢
            - Rationale:      🟢
            - Impact:         🟢
            - Remediation:    🟢
            - References:     🟢
            - Compliance:     🟢
            - Rules:          🔴 Current -> condition: all
                  rules:
                    - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\LAPS'
                    - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\LAPS -> PasswordLength'
                    - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\LAPS -> PasswordLength -> n:^(\d+) compare <= 30'
                                  Expected -> condition: all
                  rules:
                    - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\LAPS'
                    - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\LAPS -> PasswordAgeDays'
                    - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\LAPS -> PasswordAgeDays -> n:^(\d+) compare <= 30'
    </details>

  - <details><summary>18.10.42.13.1 (L1) Ensure 'Scan packed executables' is set to 'Enabled'. (Automated)
     🔴  </summary>

            - Id:             
            - Title:          🟢
            - Description:    🟢
            - Rationale:      🟢
            - Impact:         🟢
            - Remediation:    🟢
            - References:     🟢 
            - Compliance:     🟢
            - Rules:          🔴
            Current -> condition: all
                       rules:
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows Defender\Scan'
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows Defender\Scan -> DisablePackedExeScanning'
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows Defender\Scan -> DisablePackedExeScanning -> 0'            
            Proposed -> condition: any
                       rules:
                         - 'not r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows Defender\Scan'
                         - 'not r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows Defender\Scan -> DisablePackedExeScanning'
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows Defender\Scan -> DisablePackedExeScanning -> 0'
    </details>

  - <details><summary>18.10.92.2.3 (L1) Ensure 'Enable features introduced via servicing that are off by default' is set to 'Disabled'. (Automated)
     🔴  </summary>

            - Id:             
            - Title:          🟢
            - Description:    🟢
            - Rationale:      🟢
            - Impact:         🟢
            - Remediation:    🟢
            - References:     🟢 
            - Compliance:     🟢
            - Rules:          🔴
            Current -> condition: all
                       rules:
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate'
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate -> AllowTemporaryEnterpriseFeatureControl'
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate -> AllowTemporaryEnterpriseFeatureControl -> 0'           
            Proposed -> condition: any
                       rules:
                         - 'not r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate'
                         - 'not r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate -> AllowTemporaryEnterpriseFeatureControl'
                         - 'r:HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate -> AllowTemporaryEnterpriseFeatureControl -> 0'
    </details>


  - <details><summary>18.10.92.2.4 (L1) Ensure 'Remove access to \"Pause updates\" feature' is set to 'Enabled'. (Automated)
     🔴  </summary>

            - Id:             
            - Title:          🔴 Invalid character “: Expected \"
            - Description:    🟢
            - Rationale:      🟢
            - Impact:         🟢
            - Remediation:    🟢
            - References:     🟢 
            - Compliance:     🟢
            - Rules:          🟢
    </details>

  - <details><summary>18.10.56.3.9.3 (L1) Ensure 'Require use of specific security layer for remote (RDP) connections' is set to 'Enabled: SSL'. (Automated)
     🔴  </summary>

            - Id:             
            - Title:          🟢
            - Description:    🟢
            - Rationale:      🟢
            - Impact:         🔴 Invalid character “: Expected \"
            - Remediation:    🟢
            - References:     🟢 
            - Compliance:     🟢
            - Rules:          🟢
    </details>

  - <details><summary>18.10.56.3.9.4 (L1) Ensure 'Require user authentication for remote connections by using Network Level Authentication' is set to 'Enabled'. (Automated)
     🔴  </summary>

            - Id:             
            - Title:          🟢
            - Description:    🟢
            - Rationale:      🟢
            - Impact:         🔴 Invalid character “: Expected \"
            - Remediation:    🟢
            - References:     🟢 
            - Compliance:     🟢
            - Rules:          🟢
    </details>

**Comment by jk-olaoluwa** [comment]
## Update 09/09/2024

 - SCA Correction - https://github.com/wazuh/wazuh/pull/22734/commits/33619f911f2f3d487c516eb412b0d0c9d60a2cc7

**Comment by jk-olaoluwa** [comment]
# Windows 11 Enterprise Rework SCA Testing Report

## Agent running on old SCA (Windows 11 Enterprise SCA)

###  Endpoint spec

<details>
<summary>CPU/RAM</summary>

````console
> systeminfo

Host Name:                 DESKTOP-O0E8BM6
OS Name:                   Microsoft Windows 11 Pro
OS Version:                10.0.22621 N/A Build 22621
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Ooluwa@hotmail.com
Registered Organization:   N/A
Product ID:                00330-53481-88533-AAOEM
Original Install Date:     3/14/2023, 12:17:25 AM
System Boot Time:          9/12/2024, 1:11:22 AM
System Manufacturer:       LENOVO
System Model:              81X3
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 126 Stepping 5 GenuineIntel ~1298 Mhz
BIOS Version:              LENOVO ECCN46WW, 8/23/2023
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC+01:00) West Central Africa
Total Physical Memory:     16,138 MB
Available Physical Memory: 1,923 MB
Virtual Memory: Max Size:  31,498 MB
Virtual Memory: Available: 9,495 MB
Virtual Memory: In Use:    22,003 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\DESKTOP-O0E8BM6
Hotfix(s):                 5 Hotfix(s) Installed.
                           [01]: KB5042099
                           [02]: KB5012170
                           [03]: KB5043076
                           [04]: KB5041584
                           [05]: KB5043937
Network Card(s):           12 NIC(s) Installed.
                           [01]: TAP-Windows Adapter V9 for OpenVPN Connect
                                 Connection Name: Local Area Connection
                                 Status:          Media disconnected
                           [02]: Fortinet Virtual Ethernet Adapter (NDIS 6.30)
                                 Connection Name: Ethernet 2
                                 Status:          Media disconnected
                           [03]: Fortinet SSL VPN Virtual Ethernet Adapter
                                 Connection Name: Ethernet 3
                                 Status:          Media disconnected
                           [04]: Check Point Virtual Network Adapter For Endpoint VPN Client
                                 Connection Name: Ethernet 5
                                 Status:          Media disconnected
                           [05]: VMware Virtual Ethernet Adapter for VMnet1
                                 Connection Name: VMware Network Adapter VMnet1
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.92.1
                                 [02]: fe80::51a2:1533:dd2d:f6d6
                           [06]: VMware Virtual Ethernet Adapter for VMnet8
                                 Connection Name: VMware Network Adapter VMnet8
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.213.1
                                 [02]: fe80::5aa:da6d:763b:7213
                           [07]: Intel(R) Wi-Fi 6 AX201 160MHz
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.0.1
                                 IP address(es)
                                 [01]: 192.168.0.90
                                 [02]: fe80::3f94:2ac7:7c48:8700
                                 [03]: fd93:161f:ac24:0:d835:6a70:8107:7aeb
                                 [04]: 2a02:c7c:b05d:6b00:d835:6a70:8107:7aeb
                                 [05]: fd93:161f:ac24:0:789b:979e:4533:97b0
                                 [06]: 2a02:c7c:b05d:6b00:cada:bdf6:bf7d:f718
                           [08]: VirtualBox Host-Only Ethernet Adapter
                                 Connection Name: Ethernet 4
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 10.0.2.1
                                 [02]: fe80::5611:98d0:f25e:55a
                           [09]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [10]: VirtualBox Host-Only Ethernet Adapter
                                 Connection Name: Ethernet 6
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.136.2
                                 [02]: fe80::9858:997e:b835:c816
                           [11]: VirtualBox Host-Only Ethernet Adapter
                                 Connection Name: VirtualBox Host-Only Network #2
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.56.1
                                 [02]: fe80::6c7f:e476:b4f1:2328
                           [12]: VirtualBox Host-Only Ethernet Adapter
                                 Connection Name: VirtualBox Host-Only Network #3
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 169.254.87.153
                                 [02]: fe80::b4fc:c2b3:b3ae:a59a
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```
</details>

<details>
<summary>Disk</summary>

````console
>wmic logicaldisk get size,freespace,caption
Caption  FreeSpace     Size
C:       180603387904  1022870155264 
````
</details>

**Comment by ooniagbi** [comment]
SCA scan duration also reduced. LGTM!

**Comment by ooniagbi** [comment]
Merged to 4.10.0
